pkg/util/tolerations/tolerations.go

```diff
@@ -71,6 +73,10 @@ func MergeTolerations(first []api.Toleration, second []api.Toleration) []api.Tol
 			mergedTolerations = append(mergedTolerations, v1)
 		}
 	}
+	// order lexically by key
+	sort.Slice(mergedTolerations[:], func(i, j int) bool {
+		return mergedTolerations[i].Key < mergedTolerations[j].Key
```

Comment:
both Key and Effect are significant (see the key `ConvertTolerationToAMap` builds). I think it would be better to preserve the order of the passed-in lists.

If we extracted the key-building method used by `ConvertTolerationToAMap`, we could skip building `firstMap` and do this and skip sorting:
```go
for _, v := range first {
  if _, ok := secondMap[convertTolerationToKey(v)]; !ok {
    mergedTolerations = append(mergedTolerations, v)
  }
}
```

Reply:
Nice, thanks @liggitt

Reply:
@liggitt I had to modify this slightly to account for a requirement that if there are multiple toleration in the `first` slice that have conflicting keys, the last one takes precedence:

```go
// preserve order of first when merging
for _, v := range first {
  k := convertTolerationToKey(v)
  // if first contains key conflicts, the last one takes precedence
  if v, ok := firstMap[k]; ok {
    delete(firstMap, k)
    if _, ok := secondMap[k]; !ok {
      mergedTolerations = append(mergedTolerations, v)
    }
  }
}
```

Reply:
Actually, hold on, I think I can just do:

```go
// preserve order of first when merging
for _, v := range first {
  k := convertTolerationToKey(v)
  // if first contains key conflicts, the last one takes precedence
  if _, ok := secondMap[k]; !ok && firstMap[k] == v {
    mergedTolerations = append(mergedTolerations, v)
  }
}
```